### PR TITLE
Add Django models for audio

### PR DESCRIPTION
**Pull request by dhruvkb**
Addresses #75.

This PR adds models for audio, building upon the framework established by image models. Functionality that is generic has been migrated up into the base classes.

**References:** 
- https://github.com/WordPress/openverse-catalog/issues/5

### CODE REVIEW

**Comment by dhruvkb** [comment]
@krysal needs your inputs on the fields captured in the `Audio` and `AudioSet` models.

File: openverse-api/catalog/api/models/audio.py
Comment (obulat) [inline openverse-api/catalog/api/models/audio.py:62]: The sound effects in freesound have float-type duration, because some of them are less than one second in length. I'm not sure what's the best way to represent that, though. Milliseconds, maybe?

Comment (dhruvkb) [reply openverse-api/catalog/api/models/audio.py:62]: Less than one second? That's too short. Storing them in milliseconds will just add three 0s to all other sounds which are more than 1s long. But I can't think of any alternative approaches so will have to go with that.

Comment (obulat) [reply openverse-api/catalog/api/models/audio.py:62]: Tens of millisecond is also an option :)
[Some sounds](https://freesound.org/browse/tags/oh/) seem to be 0 seconds in duration on the search page, but turn out to be 0.xxx seconds when you look at the detail page.

**Comment by obulat** [comment]
Do you think it would make sense to have a `media_url` (or just `url`) column that is common for images and audios, and would hold the link to the canonical file in case of audio, and the largest size available for image? Or is it better to keep them separate, and have `url` field both on audio and image?

File: openverse-api/catalog/api/models/audio.py
Comment (dhruvkb) [reply openverse-api/catalog/api/models/audio.py:85]: Changed the default unit to milliseconds in 9da5519.

File: openverse-api/catalog/api/models/media.py
Comment (zackkrida) [inline openverse-api/catalog/api/models/media.py:167]: 😻 📝

**Review by zackkrida — APPROVED** [review]
Great work, this actually taught me a few things about the repo. ✅

File: openverse-api/catalog/api/models/audio.py
Comment (krysal) [inline openverse-api/catalog/api/models/audio.py:51]: Do we need the `filesize` of the album art for something? I don't see if it’s used for thumbnails in the case of images for example. Asking because then we could name the url of the album art as `cover_url` or `artwork_url` in that case.

**Review by krysal — APPROVED** [review]
Excellent work with documentation, much appreciated! I only have a couple of question but they are no blockers I think.
Migrations go in a separate PR?

File: openverse-api/catalog/api/models/audio.py
Comment (obulat) [inline openverse-api/catalog/api/models/audio.py:112]: Could we rewrite it using list comprehension:
`[AltAudioFile(alt_file) for alt_file in self.alt_file]` ?
It is [more Pythonic and easier to read](https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html) than map/lambda expressions.

File: openverse-api/catalog/api/models/media.py
Comment (obulat) [inline openverse-api/catalog/api/models/media.py:150]: Adding a base_url constant will probably make changing this URL easier when we need to.

File: openverse-api/catalog/api/models/audio.py
Comment (zackkrida) [reply openverse-api/catalog/api/models/audio.py:112]: TIL about list comprehensions 👍

File: openverse-api/catalog/api/models/audio.py
Comment (dhruvkb) [reply openverse-api/catalog/api/models/audio.py:51]: The filesize field is not currently being used anywhere. I was just trying to be consistent with the fields used in images, so that we can reuse code as much as possible, especially when using the same image proxy for artwork as image media.

File: openverse-api/catalog/api/models/media.py
Comment (dhruvkb) [reply openverse-api/catalog/api/models/media.py]: Addressed in 7e69204.

File: openverse-api/catalog/api/models/audio.py
Comment (dhruvkb) [reply openverse-api/catalog/api/models/audio.py]: Thanks, I love comprehensions. Resolved in 47eecf5.

**Comment by dhruvkb** [comment]
> Migrations go in a separate PR?

I was waiting for approval on the models fields before proceeding with the migrations, as frequent changes to the file would add unnecessary noise in the commits. Added migrations in c19234f.

File: openverse-api/catalog/api/models/media.py
Comment (obulat) [inline openverse-api/catalog/api/models/media.py:111]: Should we update the BASE_URL now?

**Review by zackkrida — APPROVED** [review]

**Review by obulat — APPROVED** [review]

**Review by krysal — APPROVED** [review]

File: openverse-api/catalog/api/models/media.py
Comment (zackkrida) [reply openverse-api/catalog/api/models/media.py:111]: LGTM and do it in a separate PR to rename the base url.

**Pull request merged**
